### DISCUSSION TITLE
Pico 8 multi cart

### DISCUSSION THREAD

**Discussion by dieppon** [eduardofilo/RG350_adam_image#224] [Ideas & Requests]
Is it possible to hide secondary carts from the list?

They're games that needs multiple cards but all of them appear on the list, is there a way to exclude them?

https://freds72.itch.io/virtua-racing

**Comment by eduardofilo** [comment]
The only thing I can think of is to use @neilswann80's trick of making an fbl file pointing to the first file, but that would force you to reconvert your entire cartridge collection to that format.

**Reply by dieppon** [reply]
Are there any info on how to do that? I read a line about aliases on the latest release notes but more.

**Reply by neilswann80** [reply]
### ***METHOD 1***

You could just remove either `.p8` or `.png` from the `home computers.ini` for PICO-8.
Location: `/media/data/local/home/.simplemenu/section_groups/home computers.ini`

For example: If all the individual carts end in `.png`, remove `.png` from the PICO-8 section of `home computers.ini` and rename your multi-launcher cart so it ends `.p8`.

Then only carts ending in `.p8` will appear in the SimpleMenu list, but the "invisible" carts will still be called correctly by the launcher cart.  This might be time consuming if you have a large cart collection... easy to batch rename files at the command line though, or you could use a tool like [THIS](https://www.1-4a.com/rename/)

FYI... The PICO-8 section of `home computers.ini` looks like this...
```
[PICO-8]
execs = /media/data/apps/tac08.opk
romDirs = /media/sdcard/roms/PICO8/
romExts = .png,.p8
aliasFile = /media/data/local/home/.simplemenu/alias_PICO-8.txt
```
and you'd want to edit just the `romExts` line to be:
```
romExts = .p8
```
You should be able to make all these changes from the commander app on your device... but if there's lots of renaming to be done connect you microsd to your computer and use the bulk renaming tool from the link above.

- FYI... changing from `.png` to `p8` will mean you lose the carts built-in preview image.  If you still want to see this, create a copy of the launcher cart in `PICO8/.previews` but with the '.png' extension.  The characters of the filename must be identical up to the extensions `.p8` and `.png` for the preview image to work.

**Reply by neilswann80** [reply]
### ***METHOD 2*** 
 
as @eduardofilo suggests, you could use the `fbl` shortcut trick, although I think this would be even more time consuming than the method above.

The `fbl` shortcut method would require you to create a text file for each game you want to see in your list.

Edit the text file to contain a link to a game, make sure it is the first and only line in the file; for example:
`exec=celeste.p8.png`

Then rename the text file (and the .txt extension) to 'game_name.fbl' (replace game_name with the cart/game name)
For the example above:
`Celeste.fbl`

`Celeste.fbl` is now a shortcut to `celeste.p8.png`.

Then edit the PICO-8 `home computers.ini` romExt line to read:

`romExts = .fbl`

* With this method you wouldn't have to rename any of your roms, but you'd need to create a lot of text file shortcuts!

**Comment by dieppon** [comment]
Thank you guys for all the info.

After reading all of you approaches, and knowing that my collection is mostly PNGs, I have decided to remove the `.p8` from the `romExts` line and add `.fbl`.

It worked great, all my individual carts from the virtua folder are gone and now a nice `Virtua Racing` title is on the list thanks to the `.fbl`. But there is an error when launching the file:

```
======================================
ERROR IN CART:
$$CARTLOADER.P8$$
======================================
FAILED TO OPEN CART FILE: VIRTUAL RACING.FBL
======================================
```

any ideas?

**Reply by neilswann80** [reply]
### ***METHOD 1***

- If you choose the first method I mentioned, DO NOT create any `.fbl` files.
- You'll have to follow my instructions and remove the `.png` rather than the `.p8` from `romExts`, then rename the games you want to see `.p8`.
- If you rename the individual games that are called by the multi-launch cart it will break the call command from the multi-launch cart (which is what I think you're experiencing).
i.e. You have to rename all carts other than the ones you wish to be invisible
- It will potentially  mean renaming a lot of games, which is why I provided the link to a tool for [bulk renaming.](https://www.1-4a.com/rename/)

> _*** OR ***_

### ***METHOD 2***

- If you are using the `.fbl` method  you have to remove BOTH `.p8` and `.png` from the `romExts` line and replace with `.fbl`, then create a `.fbl` file for EVERY game you want listed
i.e create `.fbl` files for all your standalone games and the multi-cart launchers, BUT NOT the games that the multi-cart launchers call.
- DO NOT rename any of the actual carts at all.

**Comment by neilswann80** [comment]
@dieppon 
I've finally had chance to properly test this game...

I downloaded the standalone zip which contained the files:

> vracing_0.p8
> vracing_1.p8
> vracing_10.p8
> vracing_11.p8
> vracing_12.p8
> vracing_2.p8
> vracing_3.p8
> vracing_4.p8
> vracing_5.p8
> vracing_6.p8
> vracing_7.p8
> vracing_8.p8
> vracing_9.p8
> vracing_main.p8
> vracing_title.p8

Renaming just `vracing_title.p8` to `Virtua Racing.png` and removing `.p8` from `romExts` worked for me!

Just make sure you rename any `.p8` standalone games to `.png` if you still want them listed and playable.

**Comment by neilswann80** [comment]
Or even easier... put everything back to how it was and use the files I've modified:
[vracing_10_standalone.zip](https://github.com/eduardofilo/RG350_adam_image/files/8220788/vracing_10_standalone.zip)

Seems that any extension name is fine, as long as you change the calls accordingly in the `main` and `title` files.
In this case I've left the launcher as `.p8` and changed the separate game files to `.p9`

**Reply by eduardofilo** [reply]
p9 :-)
Great.

**Comment by dieppon** [comment]
Awesome, thank you so much.
